### ISSUE TITLE
Add UNIX socket support to nitcorn

### ISSUE THREAD

**Issue by matthmsl** [nitlang/nit#2678]
## Description
@privat wishes to add UNIX socket support to nitcorn. This implies to add UNIX support to libevent standard library. This issue will be subject of a future PR.

## Target use
```nit
import nitcorn

# Simple Action to display the Hello World page and the get arguments
class HelloWorldAction
	super Action

	redef fun answer(http_request, turi)
	do
		var title = "Hello World!"
		var args = http_request.get_args.join(",", ":")

		var response = new HttpResponse(200)
		response.body = """
<!DOCTYPE html>
<html>
<head>
	<meta charset="utf-8">
	<title>{{{title}}}</title>
</head>
<body>
	<h1>{{{title}}}</h1>
	<p>GET args: {{{args}}}</p>
</body>
</html>"""
		return response
	end
end

# Listen on `localhost:8080`
var vh = new VirtualHost("/tmp/example.sock") # Or another class... 

# Serve `http:/hello.html` with our custom action
vh.routes.add new Route("/hello.html", new HelloWorldAction)

# Serve everything else under `http://localhost:8080/` using a `FileServer` with a root at "/var/www/"
vh.routes.add new Route(null, new FileServer("/var/www/"))

# Launch server
var factory = new HttpFactory.and_libevent
factory.config.virtual_hosts.add vh
factory.run
```

**Comment by xymus** [comment]
I would not recommend using `VirtualHost` to configure a socket. A `VirtualHost` allows different addresses to share a single socket, you probably want to keep this feature with a UNIX socket, so the `VirtualHost` should still point to an external address. Don't forget that even if the HTTP request is sent through a UNIX socket, it should have a `Host` field.

The current nitcorn API hides the *binding to TCP sockets* part. For UNIX socket I would make it explicit, so requiring the clients to write this:

~~~
...
factory.config.virtual_hosts.add vh
factory.bind_to_unix_socket("/tmp/example.sock")
factory.run
~~~

If you add the method `bind_to_unix_socket` to nitcorn, this config should *just work*. A `VirtualHost("localhost")` would receive requests from TCP and UNIX sockets.


If a server doesn't care about the HTTP host, it can set `factory.config.default_virtual_host` to catch all requests. To be more precise, the requests not handled by other virtual hosts, no matter the socket.

However, you may need to tweak nitcorn so that it doesn't automatically try to bind to TCP sockets. So maybe a virtual host without an address that would work only as the default one?

---

Fore reference, here is an HTTP request sent over a UNIX socket:

~~~
$ curl --unix-socket /tmp/my.sock http://virtualhostaddr/some/file
~~~

~~~
$ netcat -l -U /tmp/my.sock
GET /some/file HTTP/1.1
Host: virtualhostaddr
User-Agent: curl/7.52.1
Accept: */*
~~~

**Comment by privat** [comment]
Closed by #2726
